DevIncubator-Autopark/MyQueue.cs

```diff
@@ -0,0 +1,105 @@
+﻿namespace DevIncubatorAutopark
+{
+    internal class MyQueue<T>
+    {
+        private const int DefaultSize = 10;
+        private T[] queueArr;
+        private int length;
+        private int startIndex;
+        private int endIndex;
+
+        public MyQueue()
+        {
+            queueArr = new T[DefaultSize];
+        }
+
+        public MyQueue(int length)
+        {
+            if (length < 0)
+            {
+                throw new ArgumentException("Размер очереди не может быть меньше 0", nameof(length));
+            }
+            queueArr = new T[length];
+            this.length = length;
+            endIndex = 0;
+            startIndex = 0;
+        }
+
+        public MyQueue(T[] queue)
+        {
+            queueArr = queue ?? throw new ArgumentNullException(nameof(queue), "Исходные элементы очереди не могут быть null");
```

Comment:
fixed. But can you explain why bad practice?

Reply:
You copy only the reference. `queueArr` and `array` (some main collection) will have the same refferences. If the main collection changes, your collection will also change (and vice versa). But it can break logic.